External reporters?

Is there a reason a reporter cannot be referenced by an alternate javascript file? I have a use case to enable the integration of lab into Webstorm. Without going down the route of added a Java plugin, I was first going to attempt to mimic the output of the Mocha test runner in the Nodejs Plugin that ships with Webstorm. This would require a reporter that outputs something like this:
```
##teamcity[enteredTheMatrix]
##teamcity[testCount count='3']
##teamcity[testSuiteStarted nodeId='1' parentNodeId='0' name='Route' running='false' nodeType='suite' locationHint='suite://Route']
##teamcity[testSuiteStarted nodeId='2' parentNodeId='1' name='.all' running='false' nodeType='suite' locationHint='suite://Route.\.all']
##teamcity[testStarted nodeId='3' parentNodeId='2' name='should add handler' running='false' nodeType='test' locationHint='test://Route.\.all.should add handler']
##teamcity[testStarted nodeId='4' parentNodeId='2' name='should handle VERBS' running='false' nodeType='test' locationHint='test://Route.\.all.should handle VERBS']
##teamcity[testStarted nodeId='5' parentNodeId='2' name='should stack' running='false' nodeType='test' locationHint='test://Route.\.all.should stack']
##teamcity[testStarted nodeId='3' running='true']
##teamcity[testFailed nodeId='3' duration='2' message='AssertionError: expected true not to be truthy' details='    at Assertion.prop.(anonymous function) |[as ok|] (/Users/julian/dev/WebstormProjects/express/node_modules/should/lib/should.js:69:12)|n    at /Users/julian/dev/WebstormProjects/express/test/Route.js:23:34|n    at next (/Users/julian/dev/WebstormProjects/express/lib/router/route.js:90:14)|n    at /Users/julian/dev/WebstormProjects/express/test/Route.js:18:9|n    at Layer.handle |[as handle_request|] (/Users/julian/dev/WebstormProjects/express/lib/router/layer.js:76:5)|n    at next (/Users/julian/dev/WebstormProjects/express/lib/router/route.js:100:13)|n    at Route.dispatch (/Users/julian/dev/WebstormProjects/express/lib/router/route.js:81:3)|n    at Context.<anonymous> (/Users/julian/dev/WebstormProjects/express/test/Route.js:21:13)|n    at Test.Runnable.run (/Users/julian/dev/WebstormProjects/express/node_modules/mocha/lib/runnable.js:216:15)|n    at Runner.runTest (/Users/julian/dev/WebstormProjects/express/node_modules/mocha/lib/runner.js:373:10)|n    at /Users/julian/dev/WebstormProjects/express/node_modules/mocha/lib/runner.js:451:12|n    at next (/Users/julian/dev/WebstormProjects/express/node_modules/mocha/lib/runner.js:298:14)|n    at /Users/julian/dev/WebstormProjects/express/node_modules/mocha/lib/runner.js:308:7|n    at next (/Users/julian/dev/WebstormProjects/express/node_modules/mocha/lib/runner.js:246:23)|n    at Object._onImmediate (/Users/julian/dev/WebstormProjects/express/node_modules/mocha/lib/runner.js:275:5)|n    at processImmediate |[as _immediateCallback|] (timers.js:345:15)' actual='true' expected='undefined']
##teamcity[testStarted nodeId='4' running='true']
##teamcity[testFinished nodeId='4' duration='0']
##teamcity[testStarted nodeId='5' running='true']
##teamcity[testFinished nodeId='5' duration='0']
##teamcity[testSuiteFinished nodeId='2']
##teamcity[testSuiteFinished nodeId='1']
```

So, now this could possibly be a relatively specific use case and so embedding a new reporter may not be the best for this case. But then again, it could be if more people want to integrate this with Webstorm, it just feels a bit hacky. And so this is where the question comes in, can we enable the use of external reporters for a test run?

--- a/README.md
+++ b/README.md
@@ -42,6 +42,7 @@ global manipulation. Our goal with **lab** is to keep the execution engine as si
     - `tap` - TAP protocol report.
     - `lcov` - output to [lcov](http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php) format.
     - `clover` - output results in [Clover XML](https://confluence.atlassian.com/display/CLOVER) format.
+    - [Custom Reporters](#custom-reporters) - See Below
 - `-s`, `--silence` - silence test output, defaults to false.
 - `-S`, `--sourcemaps` - enables sourcemap support for stack traces and code coverage, disabled by default.
 - `-t`, `--threshold` - minimum code test coverage percentage (sets `-c`), defaults to 100%.
@@ -262,6 +263,16 @@ file to figure out where coverage is lacking.
 ```bash
 $ npm test
 ```
+
+## Custom Reporters
+
+If the value passed for `reporter` isn't included with Lab, it is loaded from the filesystem.
+If the string starts with a period (`'./custom-reporter'`), it will be loaded relative to the current working directory.
+If it doesn't start with a period (`'custom-reporter'`), it will be loaded from the `node_modules` directory, just like any module installed using `npm install`.
+
+Reporters must be a class with the following methods: `start`, `test` and `end`. `options` are passed to the class constructor upon initialization.
+
+See the [json reporter](lib/reporters/json.js) for an good starting point.
 
 ## Acknowledgements
 

--- a/lib/reporters/index.js
+++ b/lib/reporters/index.js
@@ -4,7 +4,6 @@ var Fs = require('fs');
 var Path = require('path');
 var Mkdirp = require('mkdirp');
 
-
 // Declare internals
 
 var internals = {};
@@ -20,10 +19,18 @@ internals.protos = {
     tap: require('./tap')
 };
 
+internals.requireReporter = function (reporter) {
+
+    if (reporter[0] === '.') {
+        return require(Path.join(process.cwd(), reporter));
+    }
+
+    return require(reporter);
+};
 
 exports.generate = function (options) {
 
-    var Proto = internals.protos[options.reporter];
+    var Proto = internals.protos[options.reporter] || internals.requireReporter(options.reporter);
     var reporter = new Proto(options);
 
     var dest;

--- a/package.json
+++ b/package.json
@@ -25,7 +25,8 @@
     "jslint": "0.7.x"
   },
   "devDependencies": {
-    "code": "1.x.x"
+    "code": "1.x.x",
+    "lab-event-reporter": "1.x.x"
   },
   "bin": {
     "lab": "./bin/lab"

--- a/test/reporters.js
+++ b/test/reporters.js
@@ -1313,4 +1313,47 @@ describe('Reporter', function () {
             done();
         });
     });
+
+    describe('custom reporters', function () {
+
+        it('requires a custom reporter relatively if starts with .', function (done) {
+
+            var reporter = './node_modules/lab-event-reporter/index.js';
+
+            var script = Lab.script();
+            script.experiment('test', function () {
+
+                script.test('works', function (finished) {
+
+                    finished();
+                });
+            });
+
+            Lab.report(script, { reporter: reporter }, function (err, code, output) {
+
+                expect(err).to.not.exist();
+                done();
+            });
+        });
+
+        it('requires a custom reporter from node_modules if not starting with .', function (done) {
+
+            var reporter = 'lab-event-reporter';
+
+            var script = Lab.script();
+            script.experiment('test', function () {
+
+                script.test('works', function (finished) {
+
+                    finished();
+                });
+            });
+
+            Lab.report(script, { reporter: reporter }, function (err, code, output) {
+
+                expect(err).to.not.exist();
+                done();
+            });
+        });
+    });
 });

